app/components/global-search.js

```diff
@@ -1,82 +1,78 @@
 import Ember from 'ember';
 
-const { Component, inject, run, computed } = Ember;
+const { Component, computed, inject, isEmpty, run } = Ember;
 const { service } = inject;
 const { debounce } = run;
-const { or, notEmpty } = computed;
+const { trim } = Ember.$;
+const { PromiseObject } = DS;
+
+function cleanQuery(query) {
+  return trim(query).replace(/[\-,?~!@#$%&*+\-'="]/, '');
+}
 
 export default Component.extend({
   store: service(),
   i18n: service(),
+
   classNames: ['global-search'],
-  results: [],
-  searching: false,
+
+  query: null,
+  delay: 500,
+
   showMoreInputPrompt: false,
-  showNoResultsMessage: false,
-  currentlySearchingForTerm: false,
-  hasResults: notEmpty('results'),
-  showList: or('searching', 'showMoreInputPrompt', 'showNoResultsMessage', 'hasResults'),
-  actions: {
-    clear() {
-      let input = this.$('input')[0];
-      input.value = '';
-      this.setProperties({
-        searchTerms: '',
-        showMoreInputPrompt: false,
-        showNoResultsMessage: false,
-        searching: false,
-        results: [],
-        showClearButton: false,
-      });
-    },
-    inputValueChanged(){
-      let input = this.$('input')[0];
-      let searchTerms = input.value;
-      if(this.get('currentlySearchingForTerm') === searchTerms){
-        return;
+
+  revisedQuery: computed('query', {
+    get() {
+      const query = this.get('query');
+      const revisedQuery = cleanQuery(query);
+      const validate = this.checkQueryLength(revisedQuery);
+
+      if (validate) {
+        this.set('showMoreInputPrompt', false);
+      } else {
+        this.set('showMoreInputPrompt', true);
       }
-      this.setProperties({
-        currentlySearchingForTerm: searchTerms,
-        showMoreInputPrompt: false,
-        showNoResultsMessage: false,
-        searching: false,
-      });
-      let noWhiteSpaceTerm = searchTerms.replace(/ /g,'');
-      if(noWhiteSpaceTerm.length === 0){
-        this.send('clear');
-        return;
-      } else if(noWhiteSpaceTerm.length < 3){
-        this.setProperties({
-          results: [],
-          showMoreInputPrompt: true,
+
+      return revisedQuery;
+    }
+  }).readOnly(),
+
+  _updateQuery(value) {
+    this.set('query', value);
+  },
+
+  checkQueryLength(query) {
```

Comment:
Same thing here, this is a single use function that required me to go looking aroudn for what it did.  Since what it does is super simple it can just be part of the CP.